pinns/src/pinns.c

```diff
@@ -142,20 +142,39 @@ int main(int argc, char **argv) {
     }
   } else {
     /* if we create a user namespace, we need a new process.  */
-    if (pipe2(p, O_DIRECT) < 0)
-      pexit("pipe");
+    if (socketpair(AF_UNIX, SOCK_SEQPACKET | SOCK_CLOEXEC, 0, p))
```

Comment:
Can you split this change into a separate commit?

Reply:
sure.  Pushed now